sdk/include/opentelemetry/sdk/trace/processor.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include "opentelemetry/nostd/shared_ptr.h"
+#include "opentelemetry/nostd/span.h"
+#include "opentelemetry/sdk/trace/span_data.h"
+#include "opentelemetry/trace/span.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace trace
+{
+/**
+ * Span processor allow hooks for span start and end method invocations.
+ *
+ * Built-in span processors are responsible for batching and conversion of
+ * spans to exportable representation and passing batches to exporters.
+ */
+class SpanProcessor
+{
+public:
+  virtual ~SpanProcessor() = default;
+
+  // OnStart is called when a span is started.
+  virtual void OnStart(nostd::shared_ptr<opentelemetry::trace::Span> &span) noexcept = 0;
```

Comment:
We don't mandate spans to be stored in shared_ptr's. If this needs access to a span, can it take a non-owning reference (e.g. opentelemetry::trace::Span).

Reply:
Will do.